lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -2178,6 +2476,125 @@ void GrandCentralPass::runOnOperation() {
       continue;
   }
 
+  for (auto view : views) {
+    auto bundle = view.getAugmentedType();
+
+    assert(!bundle.isRoot() && "'id' found in firrtl.view");
+
+    auto viewParentMod = view->getParentOfType<FModuleLike>();
+    auto symbolName = getModuleNamespace(viewParentMod)
+                          .newName("__" + getInterfaceName(bundle) + "__");
+
+    // Recursively walk the AugmentedBundleType to generate interfaces and XMRs.
+    // Error out if this returns None (indicating that the view is
+    // malformed in some way).  A good error message is generated inside
+    // `traverseViewBundle` or the functions it calls.
+    auto instanceSymbol =
+        hw::InnerRefAttr::get(SymbolTable::getSymbolName(viewParentMod),
+                              StringAttr::get(&getContext(), symbolName));
+    VerbatimBuilder::Base verbatimData;
+    VerbatimBuilder verbatim(verbatimData);
+    verbatim += instanceSymbol;
+    // List of interface elements.
+
+    SmallVector<VerbatimXMRbuilder> xmrElems;
+    SmallVector<InterfaceElemsBuilder> interfaceBuilder;
+
+    size_t index = 0;
+    auto ifaceName = traverseViewBundle(bundle, verbatim, xmrElems,
+                                        interfaceBuilder, view, index);
+    if (!ifaceName) {
+      removalError = true;
+      continue;
+    }
+    assert(index == view.getNumOperands());
```

Comment:
Should this be an assertion or should it be an error and pass failure? (Is this checked somewhere else?)